src/ecalc/libraries/libecalc/common/libecalc/core/models/compressor/sampled/compressor_model_sampled.py

```diff
@@ -198,6 +199,13 @@ def evaluate_rate_ps_pd(
         ps_to_evaluate = suction_pressure[indices_to_evaluate] if suction_pressure is not None else []
         pd_to_evaluate = discharge_pressure[indices_to_evaluate] if discharge_pressure is not None else []
 
+        requested_inlet_pressure = (
+            list(ps_to_evaluate) if list(ps_to_evaluate) else [math.nan] * len(list(indices_to_evaluate))
```

Comment:
not sure if I completely understand this check, supposed to be: 
```suggestion
            list(ps_to_evaluate) if ps_to_evaluate else [math.nan] * len(list(indices_to_evaluate))
```

?

Reply:
ps_to_evaluate can return an empty list, and the test is checking for that - and returns tha fallback full list of NaN instead of an empty list.

Reply:
an empty list, and None both evaluates to False (I guess not None can occur though), but it is better to just test more explicitly to understand what one expects? e.g. `if  ps_to_evaluate` or `if len(ps_to_evaluate) == 0` ...but in that case it is already a list so no need to add `list([])` ?